tsl/src/compression/compression.c

```diff
@@ -314,12 +368,54 @@ compress_chunk(Oid in_table, Oid out_table, const ColumnCompressionInfo **column
 						out_desc->natts,
 						true /*need_bistate*/);
 
-	row_compressor_append_sorted_rows(&row_compressor, sorted_rel, in_desc);
+	if (matched_index_rel != NULL)
+	{
+		DEBUG_WAITPOINT("compress_chunk_indexscan_start");
+		index_scan = index_beginscan(in_rel, matched_index_rel, SnapshotAny, 0, 0);
+		slot = table_slot_create(in_rel, NULL);
+		index_rescan(index_scan, NULL, 0, NULL, 0);
+		while (index_getnext_slot(index_scan, indexscan_direction, slot))
```

Comment:
this indexscan_direction should be based on if the matched index is itself a ASC/DESC index.